### PR TITLE
Further improvements

### PR DESCRIPTION
**Pull request by nilswagner**

### CODE REVIEW

**Comment by nschloe** [comment]
Again, line breaks.

Also, does this PR supersede PR #24?

**Comment by nilswagner** [comment]
Applied dos2unix. Multiple element types are supported.

File: meshio/permas_io.py [hunk 1]
```diff
@@ -132,59 +154,81 @@ def write(
...
+        fh.write('    $END CONSTRAINTS\n')
+        fh.write('!\n')
+        fh.write('    $LOADING NAME = LOADVAR_1\n')
+        fh.write('    $END LOADING\n')
+        fh.write('!\n')
         fh.write('$EXIT COMPONENT\n')
         fh.write('!\n')
+        fh.write('$ENTER MATERIAL\n')
+        fh.write('!\n')
+        fh.write('    $MATERIAL NAME = STEEL TYPE = ISO\n')
```
Comment (nschloe) [inline meshio/permas_io.py]: Are these leftovers from testing?

File: meshio/permas_io.py [hunk 2]
```diff
@@ -48,6 +48,19 @@ def read(filename):
                 else:
                     cells['triangle'] = [data[-3:]]
 
+        if re.search('\$ELEMENT TYPE = QUAD4', line):
+            while True:
+                line = f.readline()
+                if not line:
+                    break
+                if line.startswith('!'):
+                    break
```
Comment (nschloe) [inline meshio/permas_io.py]: Those two statements can be condensed to
```
if not line or line.startswith('!'):
```

File: meshio/permas_io.py [hunk 3]
```diff
@@ -100,6 +113,7 @@ def read(filename):
     points = numpy.reshape(points, newshape=(len(points)/3, 3))
 
     for key in cells:
+        print 'key', key
```
Comment (nschloe) [inline meshio/permas_io.py]: The reader shouldn't output anything.

File: meshio/permas_io.py [hunk 4]
```diff
@@ -132,59 +154,81 @@ def write(
...
-                    ))
+        #
+        # Avoid non-unique element numbers in case of multiple element types by num_ele !!!
+        #
+        num_ele = 0
+        
+        for meshio_type, cell in cells.iteritems():
+            numcells, num_local_nodes = cell.shape
+            permas_type = meshio_to_permas_type[meshio_type]
+            if permas_type != 'NA':
```
Comment (nschloe) [inline meshio/permas_io.py]: Better remove this line and the corresponding entry in the dictionary. If the original mesh contains vertex elements and they cannot be written in PERMAS, the user should receive notification, i.e., an error.

File: meshio/permas_io.py (hunk 1 above)
Comment (nilswagner) [inline meshio/permas_io.py]: No. I have added those lines in order to run the PERMAS Job directly. At least one material definition is needed.  Element sets are assigned to materials within the system variant.

Comment (nschloe) [inline meshio/permas_io.py]: But why these specific values? This seems to be coming from a particular application. If the format file is invalid without a MATERIAL, then we should probably name it `DUMMY_MATERIAL` to make clear that this isn't something that's meant to be used.

File: meshio/permas_io.py (hunk 4 above)
Comment (nilswagner) [inline meshio/permas_io.py]: meshio-convert should write the remaining (supported element types) afterwards. What kind of error message should be raised ?

Comment (nschloe) [inline meshio/permas_io.py]: If you just remove the `if` and the entry from the `dict`, you'll get a `KeyError` if one of the `meshio_type`s is `'vertex'`. That should be fine.

File: meshio/permas_io.py [hunk 5]
```diff
@@ -38,9 +38,7 @@ def read(filename):
         if re.search('\$ELEMENT TYPE = TRIA3', line):
```
Comment (nschloe) [inline meshio/permas_io.py]: All of the `if`s for the different element types can be condensed into one.

Comment (nilswagner) [inline meshio/permas_io.py]: How can I achieve that ?

Comment (nschloe) [inline meshio/permas_io.py]: Same way as we did the other day:
 * Create a dictionary that translates meshio types into `TRIA3`, `QUAD4` etc
 * Create the search query `'\$ELEMENT TYPE = %s' % permas_type`
 * Then, the only thing that different is the number of point references that need to be extracted (e.g., `-4`). Those can be put in the same dictionary actually, e.g.,
```python
{
  'triangle': (3, 'TRI3')
  # ...
}
```

File: meshio/permas_io.py [hunk 6]
```diff
@@ -132,59 +108,79 @@ def write(
...
-            8: 5,  # hexahedron
+        meshio_to_permas_type = {
+            'line': 'PLOTL2',
+            'triangle': 'TRIA3',
+            'quad': 'QUAD4',
+            'tetra': 'TET4',
+            'hexahedron': 'HEXE8',
+            'wedge': 'PENTA6',
+            'pyramid': 'PYRA5'
             }
```
Comment (nschloe) [inline meshio/permas_io.py]: We have almost the exact same dictionary above. This can possibly be shared.

File: meshio/permas_io.py [hunk 7]
```diff
@@ -132,59 +108,79 @@ def write(
...
+        fh.write('    $LOADING NAME = LOADVAR_1\n')
+        fh.write('    $END LOADING\n')
+        fh.write('!\n')
         fh.write('$EXIT COMPONENT\n')
         fh.write('!\n')
+        fh.write('$ENTER MATERIAL\n')
+        fh.write('!\n')
+        fh.write('    $MATERIAL NAME = DUMMY_MATERIAL TYPE = ISO\n')
+        fh.write('!\n')
+        fh.write('        $ELASTIC  GENERAL  INPUT = DATA\n')
```
Comment (nschloe) [inline meshio/permas_io.py]: If these numbers are changed to `0.0`, I think the PR is good to go.

**Pull request merged**